src/commands/howto.ts

```diff
@@ -0,0 +1,193 @@
+    .setName("howto")
+    .setDescription("HOWTOs and tutorials");
+
+  for (let subcommand of commands) {
+    rootCommandBuilder.addSubcommand((scb) =>
+      addTargetUserParam(scb.setName(subcommand.name).setDescription(subcommand.description))
+    );
+  }
+  return rootCommandBuilder.toJSON();
+};
+
+const allowedRoles = new Set(["admin", "mods", "mods+", "power-users"]);
+
+const hasSufficientPrivilege = (member: GuildMember | APIInteractionGuildMember | null) => {
+  let guildMember: GuildMember = member as GuildMember;
+  return guildMember && guildMember.roles.cache.some((role) => allowedRoles.has(role.name));
+};
+
+export const call = async (interaction: ChatInputCommandInteraction) => {
+  let invokingUser = interaction.user;
+  let targetUser = interaction.options.getUser("user", false) ?? invokingUser;
+
+  if (targetUser.bot) {
+    interaction.reply({
+      content: `${userMention(invokingUser.id)}, you cannot use this command on a bot.`,
+      ephemeral: true,
+    });
+    return;
+  }
+
+  let selfTarget = targetUser.id === invokingUser.id;
+  if (selfTarget || hasSufficientPrivilege(interaction.member)) {
+    let subCommand = interaction.options.getSubcommand();
+    let dmReply = commands.find((c) => c.name == subCommand);
+
+    if (dmReply) {
+      postHowtoDm(dmReply, targetUser).then(
+        () =>
+          interaction.reply({
+            content: `${userMention(targetUser.id)} please check your DMs`,
+            ephemeral: selfTarget,
+          }),
+        (reason: DiscordAPIError) =>
+          interaction.reply({
+            content: `Cannot send a DM to user ${targetUser.username}: ${reason.message}`,
+            ephemeral: selfTarget,
+          })
+      );
+    } else {
+      interaction.reply({
+        content: `Unknown command: \`${subCommand}\``,
+        ephemeral: true,
+      });
+    }
+  } else {
+    interaction.reply({
+      content: `${userMention(invokingUser.id)}, you are not privileged to use this command.`,
+      ephemeral: true,
+    });
+  }
+};
+
+const postHowtoDm = async (command: HowtoCommand, targetUser: User) => {
+  const setUpReactions = (msg: Message<false>) => {
+    for (let reaction of command.reactions) {
+      msg.react(reaction.emoji);
+    }
+
+    // Do not set up a collector if there are no reactions
+    if (!command.reactions.length) return;
+
+    const collector = msg.createReactionCollector({ filter: (_, reactor) => !reactor.bot });
+    collector.on("collect", (reaction, reactor) => {
+      if (reactor.bot || reactor.id != targetUser.id) return;
+
+      let emoji = reaction.emoji.name;
+      let commandName = command.reactions.find((r) => r.emoji == emoji)?.command;
+      let reactionCommand = commands.find((c) => c.name == commandName);
+      if (reactionCommand) {
+        postHowtoDm(reactionCommand, targetUser);
+      }
+    });
+  };
+
+  return targetUser.send(command.body).then(setUpReactions);
```

Comment:
You can use `await` (`const msg = await targetUser.send(command.body);`). No need to define a callback `setUpReactions`.

Reply:
Done.